scopes/react/react/react.docs.md

```diff
@@ -0,0 +1,422 @@
+
+The process of 'merging' or 'overriding' adds new properties to the existing configurations. In case of a conflict between two properties, the extension's configurations will override the extended environment's defaults.
+
+#### overrideTsConfig
+
+```ts
+overrideTsConfig(tsconfig: TsConfigSourceFile): EnvTransformer
+```
+
+Overrides the environment's default TypeScript configurations with a new ([tsconfig.json](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html)) configuration file.
+
+For example:
+
+```ts
+// ...
+
+const tsconfig = require('./typescript/tsconfig.json');
+
+export class ReactExtension {
+
+// ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overrideTsConfig(tsconfig)
+    ]);
+
+
+}
+
+// ...
+```
+
+#### overridePreviewConfig
+
+```ts
+overridePreviewConfig(config: Configuration): EnvTransformer
+```
+
+Overrides the Webpack configurations for the [Preview](/docs/environments/environment-services#preview) environment service, with a new ([webpack.config.js](https://webpack.js.org/configuration/)) configuration file.
+
+For example:
+
+```ts
+// ...
+
+const webpackConfig = require('./webpack/webpack.config');
+
+export class ReactExtension {
+
+// ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overridePreviewConfig(webpackConfig)
+    ]);
+
+
+}
+
+// ...
+```
+
+#### overrideDevServerConfig
+
+```ts
+overrideDevServerConfig(config: Configuration): EnvTransformer
+```
+
+Overrides the Webpack configurations for the [DevServer](/docs/environments/environment-services#devserver) environment service, with a new ([webpack.config.js](https://webpack.js.org/configuration/)) configuration file.
+
+For example:
+
+```ts
+// ...
+
+const webpackConfig = require('./webpack/webpack.config');
+
+export class ReactExtension {
+
+// ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overrideDevServerConfig(webpackConfig)
+    ]);
+
+
+}
+
+// ...
+```
+
+#### overrideJestConfig
+
+```ts
+overrideJestConfig(jestConfigPath: string): EnvTransformer
+```
+
+This method receives a path (as a string) to a configuration file . Overrides the default configurations for the Jest test runner with a new ([jest.config](https://jestjs.io/docs/en/configuration)) configuration file. This is done by passing the _path_ to the file as an argument.
+
+For example:
+
+```ts
+// ...
+
+export class ReactExtension {
+
+// ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    const newReactEnv = react.compose([
+      react.overrideJestConfig(require.resolve('./jest/jest.config'))
+    ]);
+
+
+}
+
+// ...
+```
+
+#### overrideBuildPipe
+
+```ts
+overrideBuildPipe(tasks: BuildTask[]): EnvTransformer
```

Comment:
there is no overrideBuildpipe at the moment

Reply:
Done. So, that means, as of now, you can only add new build tasks to an env by using the builder slot?